### ISSUE TITLE
Does not build on latest SBCL

### ISSUE THREAD

**Issue by quicklisp** [smithzvk/cl-primality#2]
When I try to build today, I get this:

```
;     (RANDOM (EXPT 2 CL-PRIMALITY::N-BITS))
; 
; caught WARNING:
;   Derived type of (SB-KERNEL:%MAKE-RATIO 1 SB-KERNEL:%DENOMINATOR) is (VALUES (AND (NOT INTEGER) (RATIONAL 0)) &OPTIONAL), conflicting with its asserted type (OR (SINGLE-FLOAT (0.0)) (DOUBLE-FLOAT (0.0d0)) (INTEGER 1)).
;   See also:
;     The SBCL Manual, Node "Handling of Types"
```

Stas tells me:

```
16:04 <stassats> it's (random (expt 2 (the fixnum x)))
16:04 <stassats> in the last function
16:05 <stassats> FIXNUM as a declaration is bad as it is
16:05 <stassats> Xach: but if the desire is still to restrict to fixnum, it (and 
      unsigned-byte fixnum) will do the trick
16:06 <stassats> while (unsigned-byte 32/64/whatever) would be better
16:07 <Bike> oh, so basically sbcl is worrying about n-bits being negative.
16:07 <stassats> yes
```

A build log is here, and has a bit more context: http://report.quicklisp.org/2015-06-04/failure-report/cl-primality.html#cl-primality

**Comment by smithzvk** [comment]
Thanks for reporting it.  I removed the offending declarations in 0ad3705 as they actually had no effect on running time anyway... just another case of run-away, fruitless, micro optimization.

However, I am unable to reproduce this error and thus cannot confirm that it is indeed fixed.  I attempted the original code against 1.2.12 and a pull from the git.code.sf.net repo master branch and don't get this error on compile or run.  Should I try this against a different branch, perhaps?

I will leave this open until I find a way to confirm it is fixed.
